### ISSUE TITLE
Training with only rgb

### ISSUE THREAD

**Issue by ahnGeo** [rvl-lab-utoronto/video_similarity_search#78]
Hello!

I wonder, are there some scripts or way training with only rgb, not including optical flow?
and do you have test top1/5 accuracy results about training ucf101 (or others) with each?

For taking optical flows of datasets(ucf101, etc...), I used extraction scripts from mmaction2. Extracted flows jpg files are divided with x and y. and it seems like they are not compatible with your code. Then, using 3d-resnet repo is the only way?

**Comment by Salarios77** [comment]
Hi! Yes, you can train with only rgb inputs (multi-view positives disabled) by setting the config parameters DATASET.POS_CHANNEL_REPLACE to False and DATASET.CHANNEL_EXTENSIONS to ''. For convenience, I've added a sample rgb-only yaml config file to [config/custom_configs/resnet_ucf_itercluster_rgbonly.yaml](https://github.com/rvl-lab-utoronto/video_similarity_search/blob/cleanup/config/custom_configs/resnet_ucf_itercluster_rgbonly.yaml) (which you can specify using the --cfg flag). You can easily modify other training hyper-parameters similarly using the yaml config file (all default parameters are specified at [config/default_params.py](https://github.com/rvl-lab-utoronto/video_similarity_search/blob/cleanup/config/default_params.py)).

We currently have an ablation on disabling the multi-view optical flow positives (i.e. only training with rgb inputs) in Table 3 of the paper for the top-1 and top-5 retrieval accuracy. Due to time constraints, we were not able to include corresponding ablations for the top-1/5 classification accuracy after finetuning. 

Regarding optical flow extraction, we take a similar approach to "Self-supervised Video Representation Learning
Using Inter-intra Contrastive Framework" [53] and only use the flow along the x direction as mentioned at the end of Section 4.1, "For multi-view positive sampling, optical flow (u) is computed using the unsupervised TV-L1 algorithm [66], with the same pre-processing procedure as in [11]. We duplicate the data from the second view to generate input clips with channel dimension 3". So, you only need to specify the path to the folder containing the optical flow for x using the config parameter DATASET.OPTICAL_U_PATH.

**Comment by ahnGeo** [comment]
Thanks for your detailed answer! 😁
